### ISSUE TITLE
L023 for CTE is unfixable with newline instead of space

### ISSUE THREAD

**Issue by juhoautio** [sqlfluff/sqlfluff#2617]
Lint & fix are both failing on this kind of pattern:

```sql
WITH cte AS
(
```

This can be fixed manually by replacing the newline after `AS` with a space, like so:

```sql
WITH cte AS (
```

It would be nice if sqlfluff can fix this automatically.

Or is this rule actually wrong? The "original query" (see below) looks ok to me.

The rule documentation for [L023](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L023) demonstrates fixing `AS(` to `AS (`. But is it correct for `lint` to require that `(` should be on the same line?

Btw the error message is confusing  because there _is_ whitespace after the `AS` keyword in my example: there's a newline. The parse tree categorizes as `newline` while there are other parts that are `whitespace`. So the technical definition seems to be that newlines are not `whitespace`. But isn't it confusing to use that kind of definition in user-facing docs? To be exact, I suppose it can be called _horizontal whitespace_.

The lint error message should be improved for this lint rule if it remains as it is, even if it becomes auto-fixable.

## Steps to Reproduce

`myquery.sql`:

```sql
WITH cte AS
(
    SELECT m_rule_id
    FROM t
)

SELECT * FROM cte;
```

### fix attempt

```
sqlfluff fix -f myquery.sql
```

```
==== finding fixable violations ====
==== no fixable linting violations found ====
All Finished!
  [1 unfixable linting violations found]
```

### lint
```
sqlfluff lint myquery.sql
```

```
== [myquery.sql] FAIL
L:   1 | P:  10 | L023 | Single whitespace expected after 'AS' in 'WITH' clause.
All Finished!
```

### parse
```
sqlfluff parse myquery.sql
```
<details>

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        with_compound_statement:
[L:  1, P:  1]      |            keyword:                                          'WITH'
[L:  1, P:  5]      |            whitespace:                                       ' '
[L:  1, P:  6]      |            common_table_expression:
[L:  1, P:  6]      |                identifier:                                   'cte'
[L:  1, P:  9]      |                whitespace:                                   ' '
[L:  1, P: 10]      |                keyword:                                      'AS'
[L:  1, P: 12]      |                newline:                                      '\n'
[L:  2, P:  1]      |                bracketed:
[L:  2, P:  1]      |                    start_bracket:                            '('
[L:  2, P:  2]      |                    [META] indent:
[L:  2, P:  2]      |                    newline:                                  '\n'
[L:  3, P:  1]      |                    whitespace:                               '    '
[L:  3, P:  5]      |                    select_statement:
[L:  3, P:  5]      |                        select_clause:
[L:  3, P:  5]      |                            keyword:                          'SELECT'
[L:  3, P: 11]      |                            [META] indent:
[L:  3, P: 11]      |                            whitespace:                       ' '
[L:  3, P: 12]      |                            select_clause_element:
[L:  3, P: 12]      |                                column_reference:
[L:  3, P: 12]      |                                    identifier:               'm_rule_id'
[L:  3, P: 21]      |                        newline:                              '\n'
[L:  4, P:  1]      |                        whitespace:                           '    '
[L:  4, P:  5]      |                        [META] dedent:
[L:  4, P:  5]      |                        from_clause:
[L:  4, P:  5]      |                            keyword:                          'FROM'
[L:  4, P:  9]      |                            whitespace:                       ' '
[L:  4, P: 10]      |                            from_expression:
[L:  4, P: 10]      |                                [META] indent:
[L:  4, P: 10]      |                                from_expression_element:
[L:  4, P: 10]      |                                    table_expression:
[L:  4, P: 10]      |                                        table_reference:
[L:  4, P: 10]      |                                            identifier:       't'
[L:  4, P: 11]      |                                [META] dedent:
[L:  4, P: 11]      |                    newline:                                  '\n'
[L:  5, P:  1]      |                    [META] dedent:
[L:  5, P:  1]      |                    end_bracket:                              ')'
[L:  5, P:  2]      |            newline:                                          '\n'
[L:  6, P:  1]      |            newline:                                          '\n'
[L:  7, P:  1]      |            select_statement:
[L:  7, P:  1]      |                select_clause:
[L:  7, P:  1]      |                    keyword:                                  'SELECT'
[L:  7, P:  7]      |                    [META] indent:
[L:  7, P:  7]      |                    whitespace:                               ' '
[L:  7, P:  8]      |                    select_clause_element:
[L:  7, P:  8]      |                        wildcard_expression:
[L:  7, P:  8]      |                            wildcard_identifier:
[L:  7, P:  8]      |                                star:                         '*'
[L:  7, P:  9]      |                whitespace:                                   ' '
[L:  7, P: 10]      |                [META] dedent:
[L:  7, P: 10]      |                from_clause:
[L:  7, P: 10]      |                    keyword:                                  'FROM'
[L:  7, P: 14]      |                    whitespace:                               ' '
[L:  7, P: 15]      |                    from_expression:
[L:  7, P: 15]      |                        [META] indent:
[L:  7, P: 15]      |                        from_expression_element:
[L:  7, P: 15]      |                            table_expression:
[L:  7, P: 15]      |                                table_reference:
[L:  7, P: 15]      |                                    identifier:               'cte'
[L:  7, P: 18]      |                        [META] dedent:
[L:  7, P: 18]      |    statement_terminator:                                     ';'
[L:  7, P: 19]      |    newline:                                                  '\n'
```

</details>

## Dialect

N/A

## Version

0.10.0

## Configuration

No custom conf

## Notes

For unit testing it's good to cover also a case where there is are least 2 CTEs so that the other doesn't have the `WITH` keyword. Example:  
```sql
WITH a AS
(
    SELECT m_rule_id
    FROM t
),

b AS
(
    SELECT m_rule_id
    FROM a
)

SELECT * FROM b;
```

**Comment by juhoautio** [comment]
I'd like to try to fix this, but I could I get some guidance on the spec? Some questions in the description.

**Comment by juhoautio** [comment]
Thanks. Feel free to assign this to me if you'd like me to work on this. As you probably know, I'm not permitted to assign anyone, even myself :)

**Comment by tunetheweb** [comment]
Yeah we don’t really use the assign functionality for that reason. Comment in the issue (like you did here) is sufficient.
